cmd/frostfs-node/config.go

```diff
@@ -968,3 +971,27 @@ func (c *cfg) configWatcher(ctx context.Context) {
 		}
 	}
 }
+
+func (c *cfg) terminateWatcher(ctx context.Context) {
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, syscall.SIGINT, syscall.SIGTERM)
+
+	select {
+	case <-ch:
+		c.closeMutex.Lock()
+
+		c.log.Info("termination signal has been received, stopping...")
+		c.ctxCancel()
+		c.setHealthStatus(control.HealthStatus_SHUTTING_DOWN)
+		for i := range c.closers {
+			c.closers[len(c.closers)-1-i]()
+		}
+		c.closers = []func(){}
+		c.log.Info("termination signal processing is complete")
+
+		c.closeMutex.Unlock()
+		return
+	case <-ctx.Done():
```

Comment:
why do we listen to `ctx` here? i would close cancel func with `terminateWatcher` and do not share it via `cfg` at all

also, it would save us from double `c.setHealthStatus(control.HealthStatus_SHUTTING_DOWN)` i guess

Reply:
@carpawell we listen here for context because we may want to exit by internal error.

Reply:
yes, and my question is: would it be better to just move listening internal error here too (in the same goroutine)? it would be the only code that is possible to cancel ctx, all the others would just wait for its closing. more straight scheme IMO (if it is possible and i have not missed something)

Reply:
Looks better, moved listening for internal errors here too.

Reply:
So can we remove `ctx.Done` now? The problem I see it that this is not a real "worker": it doesn't use context at all, but instead _manages_. Why not have this `select` executed synchronously in `wait`?